### DISCUSSION TITLE
Refactor build process for `osbuilder` and `osbuilder-tools` images

### DISCUSSION THREAD

**Discussion by jbalonso** [kairos-io/kairos#453] [Ideas]
# TL;DR:

I propose:
- ~[ ]  merging the build processes for `osbuilder` and `osbuilder-tools` into a single repository~
- [ ]  using an `Earthfile` instead of a combination `Makefile` and `Dockerfile`

# Justification

Support for `osbuilder` on arm64 is incomplete but appears unnecessarily difficult to advance.

In debugging issues around using `osbuilder` on my hybrid amd64/arm64 homelab, I found kairos-io/kairos#451 (showing a the arm64 component of a multiarch image contained an amd64 binary), ~but I could not find the source material for the culprit `osbuilder-tools` image itself to contribute a fix~. Moreover, I found kairos-io/helm-charts#5 because I needed to pin the manager to amd64 with a `nodeSelector`, but the presence of an arm64 manifest for `osbuilder-tools` shows that multiarch support was at least _intended_.

**Comment by jbalonso** [comment]
I found the `osbuilder-tools` image build process buried in an actions workflow and the `tools-image/` directory.

**Comment by mudler** [comment]
Totally agree here - the Makefile is there because it is automatically generated by operator-sdk, indeed in other controllers you might find a Makefile and an Earthfile as well. We need to consolidate Earthfiles across all controllers too, as would help also much into testing too. feel to create an issue? :slightly_smiling_face:
